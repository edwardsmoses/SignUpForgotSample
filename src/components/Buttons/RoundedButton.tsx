import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import * as Colors from '../../styles/Color';

interface RoundedButtonProps {
    onPress: () => {};
    children: React.ReactNode;
    size?: number;
}

const RoundedButton: React.FC<RoundedButtonProps> = ({
    children,
    onPress,
    size = 50,
}) => {
    return (
        <View>
            <TouchableOpacity
                onPress={onPress}
                style={[{ height: size, width: size }, styles.buttonStyle]}>
                {children}
            </TouchableOpacity>
        </View>
    );
};

export default RoundedButton;

const styles = StyleSheet.create({
    buttonStyle: {
        backgroundColor: Colors.Blue,
        borderRadius: 50,
        alignContent: 'center',
        justifyContent: 'center',
    },
});
