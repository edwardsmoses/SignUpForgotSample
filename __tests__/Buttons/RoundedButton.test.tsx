import { render, cleanup } from 'react-native-testing-library';

import React from 'react';
import { Text } from 'react-native';

import RoundedButton from '../../src/components/Buttons/RoundedButton';

afterEach(cleanup);

describe('<RoundedButton />', () => {
    it('should match snapshot', () => {
        const rendered = render(
            <RoundedButton
                onPress={() => {
                    return {};
                }}
                children={<Text>{'-'}</Text>}
            />,
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });
});
