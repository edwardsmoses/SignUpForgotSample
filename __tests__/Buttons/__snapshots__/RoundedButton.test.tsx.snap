// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundedButton /> should match snapshot 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#53a1e8",
        "borderRadius": 50,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Text>
      -
    </Text>
  </View>
</View>
`;
